geomstats/geometry/pullback_metric.py

```diff
@@ -125,34 +174,94 @@ def inner_product_derivative_matrix(self, base_point=None):
         inner_prod_deriv_mat : array-like, shape=[..., dim, dim, dim]
             Inner-product derivative matrix.
         """
-        hessian_aij = []
-        jacobian_ai = []
-        embedding_dim = self.embedding_metric.dim
-        for a in range(embedding_dim):
+        initial_ndim = base_point.ndim
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        inner_prod_deriv_mats = []
+        for point in base_point:
+            jacobian_ai = self.jacobian_immersion(point)
+            if self.dim == 1 and jacobian_ai.ndim > 2:
+                jacobian_ai = gs.squeeze(jacobian_ai, axis=-1)
 
-            def immersion_a(x):
-                return self.immersion(x)[a]
+            hessian_aij = self.hessian_immersion(point)
```

Comment:
Here we can gain performance by doing similarly to what we've done the other day: we can get the jacobian from the computation of the hessian.